pcp/PCPDynamicMeter.c

```diff
@@ -65,8 +65,8 @@ static void PCPDynamicMeter_parseMetric(PCPDynamicMeters* meters, PCPDynamicMete
       if (pmRegisterDerivedMetric(metric->name, value, &error) < 0) {
          char* note;
          xAsprintf(&note,
-                   "%s: failed to parse expression in %s at line %u\n%s\n",
-                   pmGetProgname(), path, line, error);
+                   "%s: failed to parse expression in %s at line %u\n%s\n%s",
+                   pmGetProgname(), path, line, error, pmGetProgname());
```

Comment:
Isn't this duplicating things that's already in the message string?

Reply:
Yeah - problem is the way the CRT_fatalError function behaves like popen(3), and automatically appends a colon, space and strerror(3) result.  Since the metric parse failure error message provides full context (string location of parsing failure), we follow it with a newline.

```
pcp-htop: failed to parse expression in /etc/pcp/htop/meters/entropy at line 7
Error: pmRegisterDerivedMetric("htop.entropy.avail", ...)
kernel.all.entropy.avail badness / kernel.all.entropy.poolsize * 100
                         ^
Arithmetic or relational or boolean operator expected to follow NAME

pcp-htop: Invalid argument
```

Ideally we'd not have that final line at all - it adds no value.  I guess I could create a new CRT.c function that doesn't look at errno at all?

Reply:
I don't actually mind this extra line. Was just wondering why it was there.